research-software-directory/RSD-as-a-service
# Upgrade Eslint and add Prettier

## Upgrade Eslint and add Prettier

Changes proposed in this pull request:

* Upgrade Eslint from `8.46.0` to `9.9.0` in the frontend
* Setup Prettier for the frontend

How to test:

* `docker compose down --volumes && docker compose build --parallel && docker compose up --scale data-generation=1`
* Check if everything still works
* Read the added documentation in `frontend/README.md`
* Check out the config file `frontend/eslint.config.js` (and the removed file `frontend/.eslintrc.json`)
* Check out the config files `frontend/.prettierrc` and `frontend/.prettierignore`

Closes #1198
Related to #915

PR Checklist:
